x-pack/plugins/security_solution/public/common/components/navigation/breadcrumbs/index.test.ts

```diff
@@ -42,31 +44,72 @@ const chromeMock = {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
 } as any;
 
-const mockDefaultTab = (pageName: string): SiemRouteType | undefined => {
-  switch (pageName) {
-    case 'hosts':
-      return HostsTableType.authentications;
-    case 'network':
-      return NetworkRouteType.flows;
-    case 'administration':
-      return AdministrationSubTab.endpoints;
-    default:
-      return undefined;
-  }
-};
-
 const getMockObject = (
   pageName: SecurityPageName,
   pathName: string,
   detailName: string | undefined
-): RouteSpyState & ObjectWithNavTabs => ({
-  detailName,
-  navTabs,
-  pageName,
-  pathName,
-  search: '',
-  tabName: mockDefaultTab(pageName) as HostsTableType,
-});
+): RouteSpyState & ObjectWithNavTabs => {
```

Comment:
What was the reason for the changes in and around the `x-pack/plugins/security_solution/public/common/components/navigation` folder? Do they include any functional changes or it is a refactoring?

Reply:
It's a side effect of extending `RouteSpyState` in `x-pack/plugins/security_solution/public/common/utils/route/types.ts`. It was typed in a way that all known pages could have all known tabs but it doesn't match the reality. In fact some pages can have some known tabs but some don't have tabs at al.

Another approach would be typing `tabName` in `RouteSpyState` as `string | undefined` but it looks too permissive though simpler.

I did my best to make benefits of TypeScript and simplify the code where possible by using discriminative union.